enterprise/server/remote_execution/cgroup/cgroup.go

```diff
@@ -31,6 +32,105 @@ const (
 	cidPlaceholder = "{{.ContainerID}}"
 )
 
+// min/max weight values for cgroup2 weight-based control
+const (
+	minWeight = 1
+	maxWeight = 10_000
+)
+
+// Setup configures the cgroup at the given path with the given settings.
+// IO limits are applied to the given block device, if specified.
+func Setup(ctx context.Context, path string, s *scpb.CgroupSettings, blockDevice *block_io.Device) error {
+	m, err := settingsMap(s, blockDevice)
+	if err != nil {
+		return err
+	}
+	for name, value := range m {
+		settingFilePath := filepath.Join(path, name)
+		if err := os.WriteFile(settingFilePath, []byte(value), 0); err != nil {
+			if os.IsNotExist(err) {
```

Comment:
we want to quietly fail if the cgroup doesn't exist?

Reply:
The intent was to avoid failing for on-prem setups that might not have certain cgroup controllers enabled, but in my testing we wind up seeing EPERM in this case instead of ENOENT. I changed this logic to explicitly check whether the controller is enabled and just log a warning for now if not.